test/models/location_test.rb

```diff
@@ -3,4 +3,64 @@
 
 class LocationTest < ActiveSupport::TestCase
   include TaxonomiesBaseTest
+  # List of valid location names.
+  def valid_loc_name_list
+    [
+      RFauxFactory.gen_alpha(1),
+      RFauxFactory.gen_alpha(246),
+      *RFauxFactory.gen_strings(1..246, exclude: [:html]).values,
+      RFauxFactory.gen_html(rand((1..221)))
+    ]
+  end
+
+  # List of invalid location names.
+  def invalid_loc_name_list
+    [
+      '',
+      ' ',
+      "\t",
+      *RFauxFactory.gen_strings(247).values
+    ]
+  end
+
+  test 'should create with multiple valid names' do
+    valid_loc_name_list.each do |name|
+      location = FactoryBot.build(:location, :name => name)
+      assert location.valid?, "Can't create location with valid name #{name}"
+    end
+  end
+
+  test 'should not create with multiple invalid names' do
+    invalid_loc_name_list.each do |name|
+      location = FactoryBot.build(:location, :name => name)
+      refute location.valid?, "Can create location with invalid name #{name}"
```

Comment:
it would be probably good to also verify that `assert_includes location.errors.keys, :name` so we're sure it's name attribute that is invalid (same thing you added to `test 'should not update with multiple invalid names'`